common/junit-platform-native/src/main/java/org/graalvm/junit/platform/JUnitPlatformFeature.java

```diff
@@ -93,6 +96,46 @@ public void beforeAnalysis(BeforeAnalysisAccess access) {
         Launcher launcher = LauncherFactory.create();
         TestPlan testplan = discoverTestsAndRegisterTestClassesForReflection(launcher, selectors);
         ImageSingletons.add(NativeImageJUnitLauncher.class, new NativeImageJUnitLauncher(launcher, testplan));
+
+        Method registerAllDeclaredMethods = findMethodOrNull(RuntimeReflection.class, "registerAllDeclaredMethods", Class.class);
```

Comment:
The required `RuntimeReflection#registerAllDeclaredMethods` method was introduced in `graalvm-23.0`. However, since the `native-build-tools` are currently utilizing `graalvm-22.3.5`, we employ the method reflectively only when it is available.

Reply:
@fniephaus Could you assist me in identifying additional reviewers?
Thank you!

Reply:
@tzezula actually, `native-build-tools` are currently using graalvm `23.0.2`. This change was introduced in [this PR](https://github.com/graalvm/native-build-tools/pull/565) and it is a part of [our latest release](https://github.com/graalvm/native-build-tools/releases/tag/0.10.1) .

Reply:
Perfect, I will rebase the PR and remove the reflection.
Thanks!